### ISSUE TITLE
Adjusting Detailer Resolution-Blurry Faces with SDXL

### ISSUE THREAD

**Issue by ckao10301** [ltdrdata/ComfyUI-Impact-Pack#419]
The inpainted faces are blurry. I want the resolution of the face to be 1024x1024. I set the guide size to 1024 already-any idea why it's not working? I'm using SDXL with controlnet. SD 1.5 works fine somehow-are there different settings I should be using for SDXL? I've tried many SDXL models and none of them work well.

Another question-is there documentation that explains what "noise mask" does?

<img width="964" alt="image" src="https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/129625358/146da7b6-ce9d-4fc1-a87b-5a6c98c75308">

I'm hoping the inpainted face can be very detailed, such as this photo:
![image](https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/129625358/d465870e-e7fa-44c2-9bc7-04d6c89513f4)

**Comment by ltdrdata** [comment]
Your reference image is 1344x1728. But you restricted `guide_size` as 1024 in `crop_factor` 3.0.
And the guide_size_for is set as `crop_region`.
 detailer hook.
It means the face portion is much smaller than it.

https://youtu.be/KvZ8ucBqyqw?feature=shared

The name "noise_mask" is derived from SetLatentNoiseMask, determining whether to apply a mask during inpainting or only when pasting. If the noise_mask is enabled, the mask will be applied during inpainting; if it is disabled, the mask will not be applied during inpainting.

What you should pay attention to is raising the guide_size(and max_size) and the Detailer Hook.
https://youtu.be/KE-2Kxy9NdQ?feature=shared

**Comment by ckao10301** [comment]
I changed crop factor to 1, guide_size_for to bounding box, and guide_size_ to 1024.  The log shows that the crop region is 175x236 upscaled to 1024, 1380. But the face is still blurry. Both the "
cropped refined" photo and the final photo have blurry faces. Shouldn't upscaling add more detail to the face? How can I make it add more detail and skin texture to the face while upscaling?

<img width="825" alt="image" src="https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/129625358/26e968b7-f830-491e-aaab-7c82b7fc5eb0">

<img width="855" alt="image" src="https://github.com/ltdrdata/ComfyUI-Impact-Pack/assets/129625358/ee3c8250-609c-4a80-a3a5-308224f695e1">

**Comment by ltdrdata** [comment]
The representation of skin texture is crucial and highly dependent on the capabilities of the `model` and the `prompt` used. You should use a model that has the ability to capture such textures. And have to use proper texture prompt.
It should also be possible to apply LoRa.

And the detailer hook introduced in the second YouTube link provides noise that can be used as a material to add detail.

Basically, the Detailer is an inpainting process that takes place in an enlarged image.

**Comment by ckao10301** [comment]
Ok those are all good tips, thanks.

I didn't quite understand the explanation:
"The name "noise_mask" is derived from SetLatentNoiseMask, determining whether to apply a mask during inpainting or only when pasting. If the noise_mask is enabled, the mask will be applied during inpainting; if it is disabled, the mask will not be applied during inpainting."

What does it mean, do apply mask during pasting? Why is a mask needed while pasting the cropped image into the original image? If there is no mask during inpainting, then inpainting wouldn't be possible because the original image wouldn't be changed. What am I misunderstanding?

**Comment by ltdrdata** [comment]
If a mask is not applied during inpainting, the entire crop_region will be inpainted. When denoise is low, the composition doesn't change significantly, so pasting it into the original image without a mask doesn't pose a big issue.

During the VAE encode/decode and resize processes, slight distortion occur, so it's crucial not to touch areas outside the mask that must be updated. This is why the mask is used when pasting only the mask area from the inpainted image into the original image.Furthermore, the boundary between the inpainted area and the original area is visible, so feathering is applied to mitigate this.

**Comment by ckao10301** [comment]
Great, explanation thanks!
